Mathlib/CategoryTheory/Category/Cat/Adjunction.lean

```diff
@@ -51,4 +49,48 @@ def typeToCatObjectsAdj : typeToCat ⊣ Cat.objects where
           obtain rfl := Discrete.eq_of_hom f
           aesop_cat ) }
 
+
+/-- The connected components functor -/
+def connectedComponents.{v} : Cat.{v, u} ⥤ Type u where
+  obj C := ConnectedComponents C
+  map F :=
+    Quotient.lift (Quotient.mk (Zigzag.setoid _) ∘ F.obj)
+      (fun _ _ ↦ Quot.sound ∘ zigzag_obj_of_zigzag F)
+  map_id _ := funext fun x ↦ (Quotient.exists_rep x).elim (fun _ h ↦ by simp [<- h]; rfl)
+  map_comp _ _ := funext fun x ↦ (Quotient.exists_rep x).elim (fun _ h => by simp [<- h])
+
+
+private def fnToFctr  {X C} (fct : connectedComponents.obj C ⟶ X) : (C ⥤ typeToCat.obj X) where
+  obj :=  Discrete.mk ∘ fct ∘ Quotient.mk (Zigzag.setoid _)
+  map :=  Discrete.eqToHom ∘ congrArg fct ∘ Quotient.sound ∘ Zigzag.of_hom
+
+private def fctrToFn {X} {C : Cat} (fctr :C ⥤ typeToCat.obj X)  : (connectedComponents.obj C ⟶ X) :=
+  Quotient.lift
+    (fun c => (fctr.obj c).as)
+    (fun _ _ h => eq_of_zigzag X (zigzag_obj_of_zigzag fctr h))
+
+/-- `typeToCat : Type ⥤ Cat` is right adjoint to `connectedComponents : Cat ⥤ Type` -/
+def connectedComponentsTypeToCatAdj : connectedComponents ⊣ typeToCat where
+  homEquiv C X := {
+    toFun := fnToFctr
+    invFun := fctrToFn
+    left_inv := fun f ↦ funext fun x ↦ by
+      obtain ⟨x, h⟩ := Quotient.exists_rep x
+      rw [← h]
+      rfl
+    right_inv  := fun fctr ↦
+      Functor.hext (fun _ ↦ rfl) (fun c d f ↦
+        have : Subsingleton (fctr.obj c ⟶ fctr.obj d) := Discrete.instSubsingletonDiscreteHom _ _
+        (Subsingleton.elim (fctr.map f) _).symm.heq) }
+  unit := { app:= fun C  ↦ fnToFctr (𝟙 _) }
+  counit :=  {
+      app := fun X => fctrToFn (𝟙 typeToCat.obj X)
+      naturality := fun _ _ _ =>
+        funext (fun xcc => by
+          obtain ⟨x,h⟩ := Quotient.exists_rep xcc
+          aesop_cat) }
+  homEquiv_unit := fun {C X h} => Functor.hext (fun _ => by rfl) (fun _ _ _ => by rfl)
+  homEquiv_counit := fun {C X G} => by funext cc;obtain ⟨_,_⟩ := Quotient.exists_rep cc; aesop_cat
```

Comment:
```suggestion
  homEquiv_counit := fun {C X G} => by funext cc;obtain ⟨_,_⟩ := Quotient.exists_rep cc; aesop_cat
```

Reply:
You should implement this suggestion, automation takes care of the field `homEquiv_unit` so it can be safely removed.